sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.resolved) {
```

Comment:
@cloud-fan Sorry, I noticed that I made a mistake. The new code should be ``val nullable = !encoder.clsTag.runtimeClass.isPrimitive``. It worked well.
I will commit this code.